packages/jest-environment-jsdom/src/index.js

```diff
@@ -18,15 +18,20 @@ const ModuleMocker = require('jest-mock');
 
 class JSDOMEnvironment {
 
+  console: Object;
   document: ?Object;
   fakeTimers: ?FakeTimers;
   global: ?Global;
   moduleMocker: ?ModuleMocker;
 
-  constructor(config: Config): void {
+  constructor(config: Config, createJestConsole: (x: number) => Object): void {
```

Comment:
I was more thinking of the higher level context being dependent on `env.console`. This change would require TestEnvironments to have a `.console.getBuffer()` if they want their output included with the results.